src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java

```diff
@@ -293,20 +343,92 @@ public Response downloadZip(JsonObject content,
       }
    }
 
-   protected void validateCommand(String commandName)
+   @POST
+   @javax.ws.rs.Path("/commands/{commandName}/execute")
+   @Consumes(MediaType.APPLICATION_JSON)
+   public Response executeCommandJson(JsonObject content,
+            @PathParam("commandName") @DefaultValue(DEFAULT_COMMAND_NAME) String commandName,
+            @Context HttpHeaders headers)
+            throws Exception
+   {
+      exposedCommands.validateCommand(commandName);
+      try (CommandController controller = getCommand(commandName, headers))
+      {
+         helper.populateControllerAllInputs(content, controller);
+         if (controller.isValid())
+         {
+            Result result = controller.execute();
+            if (result instanceof Failed)
+            {
+               JsonObjectBuilder builder = Json.createObjectBuilder();
+               helper.describeResult(builder,result);
+               return Response.status(Status.INTERNAL_SERVER_ERROR).entity(builder).build();
+            }
+            else
+            {
+               String entity = getMessage(result);
+               String contentType = MediaType.TEXT_PLAIN;
+               if (isJsonString(entity)) {
+                  contentType = MediaType.APPLICATION_JSON;
+               }
+               return Response
+                        .ok(entity)
+                        .type(contentType)
+                        .build();
+            }
+         }
+         else
+         {
+            JsonObjectBuilder builder = createObjectBuilder();
+            helper.describeValidation(builder, controller);
+            return Response.status(Status.PRECONDITION_FAILED).entity(builder.build()).build();
+         }
+      }
+   }
+
+   /**
+    * Returns the result message handling composite results
+    */
+   protected static String getMessage(Result result)
+   {
+      if (result instanceof CompositeResult) {
+         CompositeResult compositeResult = (CompositeResult) result;
+         StringBuilder builder = new StringBuilder();
+         List<Result> results = compositeResult.getResults();
+         for (Result child : results)
+         {
+            String message = getMessage(child);
+            if (message != null && message.trim().length() > 0)
+            {
+               if (builder.length() > 0)
+               {
+                  builder.append("\n");
+               }
+               builder.append(message);
+            }
+         }
+         return builder.toString();
+
+      }
+      return result.getMessage();
+   }
+
+   private boolean isJsonString(String text)
```

Comment:
My worry with that is if the `result.message` is already a JSON string, it'll just wrap the JSON as a String right? So callers will have to parse JSON, find the message string then parse JSON again.

Maybe a cleaner solution is we look for an `attributeMap` entry for the `mediaType` and `entity` - if they are there then use those, otherwise just return the message text? 

Then for commands that want to expose a POJO as JSON/YAML/XML they store the POJO in the attribute map along with its default media type - and the JAXRS method does the conversion of POJO to JSON/YAML/XML? I guess the downside is that'd not work for `CompositeResult` instances. But then this is all really aimed at simple single page commands using `GET` after all?